aegisthus-hadoop/src/main/java/com/netflix/aegisthus/io/writable/AegisthusKeyMapper.java

```diff
@@ -1,13 +1,49 @@
 package com.netflix.aegisthus.io.writable;
 
+import com.netflix.Aegisthus;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.BytesType;
+import org.apache.cassandra.db.marshal.TypeParser;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.exceptions.SyntaxException;
+import org.apache.cassandra.serializers.MarshalException;
 import org.apache.hadoop.mapreduce.Mapper;
 
 import java.io.IOException;
 
 public class AegisthusKeyMapper extends Mapper<AegisthusKey, AtomWritable, AegisthusKey, AtomWritable> {
+    private AbstractType<?> rowKeyComparator;
+    
+    @Override protected void setup(
+            Context context)
+            throws IOException, InterruptedException {
+        super.setup(context);
+
+        String rowKeyType = context.getConfiguration().get(Aegisthus.Feature.CONF_KEYTYPE, "BytesType");
+
+        try {
+            rowKeyComparator = TypeParser.parse(rowKeyType);
+        } catch (SyntaxException | ConfigurationException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     @Override
     protected void map(AegisthusKey key, AtomWritable value, Context context)
             throws IOException, InterruptedException {
-        context.write(key, value);
+        try {
+            String[] keyWhitelist = context.getConfiguration().getStrings(Aegisthus.Feature.CONF_KEY_WHITELIST);
+            if (keyWhitelist != null && keyWhitelist.length > 0) {
+                String readableKey = rowKeyComparator.getString(key.getKey());
+                boolean matches = false;
+                for (String prefix : keyWhitelist) {
+                    if (readableKey.startsWith(prefix + "\\:")) matches = true;
```

Comment:
It would need to be a "context.write" and then a return, which would mean
duplicating the context.write call, so it'd work out to the same number of
lines.

On Thu, Apr 7, 2016 at 4:42 PM, Daniel Chia <notifications@github.com>
wrote:

> In
> aegisthus-hadoop/src/main/java/com/netflix/aegisthus/io/writable/AegisthusKeyMapper.java
> <https://github.com/coursera/aegisthus/pull/9#discussion_r58962759>:
>
> >      @Override
> >      protected void map(AegisthusKey key, AtomWritable value, Context context)
> >              throws IOException, InterruptedException {
> > -        context.write(key, value);
> > +        try {
> > +            String[] keyWhitelist = context.getConfiguration().getStrings(Aegisthus.Feature.CONF_KEY_WHITELIST);
> > +            if (keyWhitelist != null && keyWhitelist.length > 0) {
> > +                String readableKey = rowKeyComparator.getString(key.getKey());
> > +                boolean matches = false;
> > +                for (String prefix : keyWhitelist) {
> > +                    if (readableKey.startsWith(prefix + "\\:")) matches = true;
>
> perhaps you don't not need matches, and just return here on match?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/coursera/aegisthus/pull/9/files/3d63589ee9f5e6bea505d8514bdee28229c0b281#r58962759>
>